### PR TITLE
MGMT-7528: implement V2 download cluster files API

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
It will not include downloading discovery.ign file,
which will be implemented in a dedicated endpoint

# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -5641,3 +5642,35 @@ func (b *bareMetalInventory) V2DownloadInfraEnvFiles(ctx context.Context, params
...
+
+func (b *bareMetalInventory) V2DownloadClusterFiles(ctx context.Context, params installer.V2DownloadClusterFilesParams) middleware.Responder {
+	if funk.Contains(clusterPkg.ClusterOwnerFileNames, params.FileName) {
+		authPayload := ocm.PayloadFromContext(ctx)
+		if ocm.UserRole != authPayload.Role {
+			errMsg := fmt.Sprintf("File '%v' is accessible only for cluster owners", params.FileName)
+			payload := common.GenerateInfraError(http.StatusForbidden, errors.New(errMsg))
+			return installer.NewV2DownloadClusterFilesForbidden().WithPayload(payload)
+		}
+	}
```
Comment (avishayt) [inline internal/bminventory/inventory.go:5654]: We removed kubeconfig and admin-password from this API, so this code is no longer necessary.

Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go]: fixed

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -5641,3 +5642,27 @@ func (b *bareMetalInventory) V2DownloadInfraEnvFiles(ctx context.Context, params
...
+func (b *bareMetalInventory) V2DownloadClusterFiles(ctx context.Context, params installer.V2DownloadClusterFilesParams) middleware.Responder {
+	respBody, contentLength, err := b.V2DownloadClusterFilesInternal(ctx, params)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	return filemiddleware.NewResponder(installer.NewV2DownloadClusterFilesOK().WithPayload(respBody), params.FileName, contentLength)
+}
+
+func (b *bareMetalInventory) V2DownloadClusterFilesInternal(ctx context.Context, params installer.V2DownloadClusterFilesParams) (io.ReadCloser, int64, error) {
+
```
Comment (avishayt) [inline internal/bminventory/inventory.go:5655]:
```suggestion
```

Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go]: fixed

**Review by avishayt — APPROVED** [review]

**Review by avishayt — APPROVED** [review]

**Pull request merged**